extide/gradle/src/org/netbeans/modules/gradle/loaders/LegacyProjectLoader.java

```diff
@@ -380,8 +380,10 @@ private static String getLocation(Throwable locationAwareEx) {
             return (String)locationAccessor.invoke(locationAwareEx);
         } catch (ReflectiveOperationException ex) {
             LOG.log(Level.FINE,"Error getting location", ex);
-            return null;
+        } catch (IllegalArgumentException iae) {
+            LOG.log(Level.FINE,"Trying to get location from an exception, that has no location property: " + locationAwareEx.getClass().getName());
         }
+        return null;
```

Comment:
the problem might be deeper than this. The method accepts throwable but expects it to be `LocationAwareException`. If someone would call it with the wrong exception type 
```java
            if (locationAccessor == null) {
                locationAccessor = locationAwareEx.getClass().getMethod("getLocation"); // NOI18N
            }
```
would intitialize the field wrong if it happens to have that method, and all subsequent calls will fail or do really weird things.

I personally am not a fan of `Throwable` usage, since it is in 99% of all cases not something you should be catching.

Reply:
Well, I think it's a tradeoff we need to pay when we try to cover multiple versions at once. My mind could trick me, but some exceptions had `getLocation` methods even before `LocationAwareException` was introduced.

This is just some heuristic, trying to figure out what's behind a Gradle build failure. If we can get a location out of that we display it as an editor hint on the source code.

Well, if the heuristic does not work we just shall let it go...

Reply:
but doesn't the `Method` field has to fit to the right class? If this method is supposed to be as generic as possible and work for all exceptions which have a `getLocation` method, I don't think it should be cached as field. It should be checked if that exception has `getLocation` on every invocation:

```java
            Method locationAccessor = locationAwareEx.getClass().getMethod("getLocation"); // NOI18N
            return (String)locationAccessor.invoke(locationAwareEx);
```

Reply:
this illustrates the issue:
```java
        Integer five = 5;
        Double pi = Math.PI;
        
        Method stringAccessor =  five.getClass().getMethod("toString");
        System.out.println(stringAccessor.invoke(pi));
```

Reply:
Point taken. Haven't really checked what's in the try block. Thanks!